compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/builders/ExpressionHelpers.kt

```diff
@@ -179,26 +182,44 @@ fun IrBuilderWithScope.irEqeqeq(arg1: IrExpression, arg2: IrExpression) =
     context.eqeqeq(startOffset, endOffset, arg1, arg2)
 
 fun IrBuilderWithScope.irNull() =
-    IrConstImpl.constNull(startOffset, endOffset, context.irBuiltIns.nothingNType)
+    irNull(context.irBuiltIns.nothingNType)
+
+fun IrBuilderWithScope.irNull(irType: IrType) =
+    IrConstImpl.constNull(startOffset, endOffset, irType)
 
 fun IrBuilderWithScope.irEqualsNull(argument: IrExpression) =
-    primitiveOp2(
-        startOffset, endOffset, context.irBuiltIns.eqeqSymbol, context.irBuiltIns.booleanType, IrStatementOrigin.EQEQ,
-        argument, irNull()
-    )
+    irEquals(argument, irNull())
 
 fun IrBuilderWithScope.irEquals(arg1: IrExpression, arg2: IrExpression, origin: IrStatementOrigin = IrStatementOrigin.EQEQ) =
     primitiveOp2(
         startOffset, endOffset, context.irBuiltIns.eqeqSymbol, context.irBuiltIns.booleanType, origin,
         arg1, arg2
     )
 
-fun IrBuilderWithScope.irNotEquals(arg1: IrExpression, arg2: IrExpression) =
-    primitiveOp1(
-        startOffset, endOffset, context.irBuiltIns.booleanNotSymbol, context.irBuiltIns.booleanType, IrStatementOrigin.EXCLEQ,
-        irEquals(arg1, arg2, origin = IrStatementOrigin.EXCLEQ)
+fun IrBuilderWithScope.irNot(arg: IrExpression, origin: IrStatementOrigin) =
+    primitiveOp1(startOffset, endOffset, context.irBuiltIns.booleanNotSymbol, context.irBuiltIns.booleanType, origin, arg)
+
+fun IrBuilderWithScope.irOrOr(arg1: IrExpression, arg2: IrExpression) =
+    irIfThenElse(
+        context.irBuiltIns.booleanType,
+        arg1,
+        irTrue(),
+        arg2,
+        IrStatementOrigin.OROR
+    )
+
+fun IrBuilderWithScope.irAndAnd(arg1: IrExpression, arg2: IrExpression) =
+    irIfThenElse(
+        context.irBuiltIns.booleanType,
+        arg1,
+        arg2,
+        irFalse(),
+        IrStatementOrigin.ANDAND
```

Comment:
In this case I want to have short-circuit behavior. Is there any use case where we might want to disable it?

Reply:
My concern is if someone else will use it in future whether they will realize this will be short circuited. But since all ANDAND origin have short circuited behavior, it should be fine.